lib/resty/counter.lua

```diff
@@ -15,13 +20,14 @@ local function sync(_, self)
   local err, _
   local ok = true
   for k, v in pairs(self.increments) do
-    self.increments[k] = nil
     _, err, _ = self.dict:incr(k, v, 0)
     if err then
       ngx.log(ngx.WARN, "error increasing counter in shdict key: ", k, ", err: ", err)
       ok = false
     end
   end
+
+  clear_tab(self.increments)
```

Comment:
Thanks for the PR!

I have one question though: since it's possible to yield in dict:incr, the value in self.increments might have been modified in another request after `incr()` and before `clear_tab`. So there might be a race condition that some data get lost.

Reply:
> since it's possible to yield in `dict:incr`,

you are wrong. you can take a look at the source code, I'm sure there was no `yield` in `dict:incr`.

https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/shdict.lua#L444

The lock (shmtx_lock) can be triggered inside [ngx_lua_ffi_shdict_incr](https://github.com/openresty/lua-resty-core/blob/master/lib/resty/core/shdict.lua#L501) , which is very different from Lua`yield`.

Reply:
👍 you are right, i was thinking something else. this looks good to me then.

let's add a compat mode to lua interpreters without table.clear as well (similar to https://github.com/Kong/kong/blob/master/kong/router.lua#L48)

Reply:
ok, fixed already